### ISSUE TITLE
No CSS in Netlify or CircleCI production build (empty "gatsby-inlined-css"), but works perfectly in develop or when I manually deploy build to AWS S3 from command line awscli

### ISSUE THREAD

**Issue by drake-smith** [gatsbyjs/gatsby#6233]
As mentioned above, my Gatsby x Contentful website CSS builds successfully and looks as intended locally in dev or when I deploy the build to Amazon AWS manually via command line (aws cli) (I set up a yarn deploy script). However, any build in Netlify and CircleCI is likewise successful BUT there is no CSS. <style id="gatsby-inlined-css"> is empty. My guess is likely to be 'gatsby-plugin-sass' not working as intended. 

There have been other issues posted here with similar problems, and I've seen some folks say they fixed it by force updating gatsby-plugin-sass to version 1.0.19. I've tried this to no avail. I've also checked out my image imports to make sure I didn't accidentlally capitilizae a file name (i.e. import 'HeroBanner' instead of 'heroBanner' from '../images/hero-banner.jpg') which I saw could cause this. I'm totally stuck and don't know what to do. Any advice on the steps I can take to rectify this?

Current versions:

"gatsby": "^1.9.251",
    "gatsby-link": "^1.6.40",
    "gatsby-plugin-algolia": "^0.0.3",
    "gatsby-plugin-create-client-paths": "^1.0.8",
    "gatsby-plugin-lodash": "^1.0.11",
    "gatsby-plugin-react-helmet": "^2.0.10",
    "gatsby-plugin-sass": "1.0.19",
    "gatsby-source-contentful": "^1.3.48",

**Comment by m-allanson** [comment]
If you can build the site locally but not in other environments, that suggests a difference in the environments - maybe some missing or different dependencies? Can you try both of these and see if the site builds correctly:

**1**

- assuming your site is on GitHub (or a similar service)
- check out your site into a new directory on your machine
- run `yarn`
- run `gatsby build`
- check built output

**2**

- delete your `node_modules` directory
- delete your `yarn.lock` directory
- run `yarn`
- run `gatsby build`
- check built output

**Comment by Polcius** [comment]
Same happens to me. When I run 'gatbsy develop' I can see the correct CSS I intended.

But when I run 'gatsby build' or I build to Netlify, the CSS is messed up. I don't know at which step, Gatsby is adding some unwanted CSS at the build process.

**Comment by Polcius** [comment]
@drake-smith I see. I am using Windows 10. I am gonna try to do more tests this afternoon. In my case styles are not empty, but some unwanted styles are overwriting the ones I intend.

**Comment by KyleAMathews** [comment]
The linix file system works differently than Mac/windows in that file name casing matters. A-file.js is different than a-file.js. I'd try building your site in a linux environment e.g. a docker container or VM and see if the same problem mainifests.

**Comment by m-allanson** [comment]
Note that Netlify provide a Docker build image so you can replicate the environment they use for builds: https://github.com/netlify/build-image

**Comment by wimbush** [comment]
Hi, just wanted to chime in on this one. 

I am using gatsby-plugin-sass and css modules.

The problem occurs after upgrading v1 to v2.

The problem of no styles is present both when running gatsby build and on Netlify. 

Previous advice to lock down gatsby-plugin-sass to 1.0.19, does not work. 

Plugin version 1.0.19 produces an error when running gatsby develop:

- The plugin "gatsby-plugin-sass@1.0.19" is exporting a variable named "modifyWebpackConfig" which isn't an API.

I can't seem to find a working plugin version.

Thus far I have not solved the issue. 

Please let me know if I can provide more details.

**Comment by wimbush** [comment]
In my case, this problem existed using gatsby build on a local Mac, as well as netlify. The styles are not present. 

I believe possibly related to gatsby-plugin-sass and css modules.

**Comment by scottweston** [comment]
Double check you are importing the root of your scss (e.g. `import '../assets/scss/main.scss'`) somewhere common in your components/pages. I did some refactoring recently to move to `@reach/router` and in doing so ended up with a split tree in my components hierarchy. I don't know why but in `develop` mode I got the correct styles over the entire site however on a `build` most pages had no styles at all. The thing I noticed was that one of my components for the pages with styles were importing my `main.scss` file but I'd removed that component from my refactored `<Router>`ed pages.

**Comment by drake-smith** [comment]
@DSchau Yes, sorry, was writing this up but I should have posted this before closing the issue.

The culprit was a font file with a capital letter in it. Of course, I had quadruple-checked my Sass file names but not the font files.

I would be interested in having the ability to use Pascal Case (i.e. TextBlock.js) in my Gatsby project that's hosted on Netlify am not sure how to approach that.

**Comment by ariley** [comment]
I have the same problem. Now my site broke twice and had to revert, can't find what breaks it.

**Comment by ariley** [comment]
Problem solved: missing ‘_astro’ folder in the app.yaml file.
